agent/engine/docker_client.go

```diff
@@ -1038,6 +1043,99 @@ func (dg *dockerGoClient) createVolume(ctx context.Context,
 	return volumeResponse{Volume: volume, Error: nil}
 }
 
+func (dg *dockerGoClient) InspectVolume(name string, timeout time.Duration) volumeResponse {
+	// Create a context that times out after the 'timeout' duration
+	// Injecting the 'timeout' makes it easier to write tests.
+	// Eventually, the context should be initialized from a parent root context
+	// instead of TODO.
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+
+	// Buffered channel so in the case of timeout it takes one write, never gets
+	// read, and can still be GC'd
+	response := make(chan volumeResponse, 1)
+	go func() { response <- dg.inspectVolume(ctx, name) }()
+
+	// Wait until we get a response or for the 'done' context channel
+	select {
+	case resp := <-response:
+		return resp
+	case <-ctx.Done():
+		// Context has either expired or canceled. If it has timed out,
+		// send back the DockerTimeoutError
+		err := ctx.Err()
+		if err == context.DeadlineExceeded {
+			return volumeResponse{Volume: nil, Error: &DockerTimeoutError{timeout, "inspecting volume"}}
+		}
+		// Context was canceled even though there was no timeout. Send
+		// back an error.
+		return volumeResponse{Volume: nil, Error: &CannotInspectVolumeError{err}}
```

Comment:
Is there any reason to make this error a reference here, but an object in `inspectVolume`? Otherwise we should be consistent here.

Reply:
made them all references, being consistent with other functions in the client.